lib/ProductOpener/Display.pm

```diff
@@ -5146,133 +5115,77 @@ sub search_and_display_products ($request_ref, $query_ref, $sort_by, $limit, $pa
+			$log->debug("Counting MongoDB documents for query", {query => $query_ref}) if $log->is_debug();
+			# test if query_ref is empty
+			if (single_param('no_count')) {
+				# Skip the count if it is not needed
+				# e.g. for some API queries
+				$log->debug("no_count is set, skipping count") if $log->is_debug();
+			}
+			elsif (keys %{$query_ref} > 0) {
+				#check if count results is in cache
+				my $key_count = generate_cache_key("search_products_count", $query_ref);
+				$log->debug("MongoDB query key - search_products_count", {key => $key_count}) if $log->is_debug();
+				my $results_count = get_cache_results($key_count, $request_ref);
+				if (not defined $results_count) {
+
+					$log->debug("count not in cache for query", {key => $key_count}) if $log->is_debug();
+
+					# Count queries are very expensive, if possible, execute them on the postgres cache
+					if (can_use_query_cache()) {
+						$count = execute_count_tags_query($query_ref);
 					}
-				);
-				$log->info("MongoDB count query ok", {error => $@, count => $count}) if $log->is_info();
 
-				my $aggregate_parameters
-					= [{"\$match" => $query_ref}, {"\$sample" => {"size" => $request_ref->{sample_size}}}];
-				$log->debug("Executing MongoDB query", {query => $aggregate_parameters}) if $log->is_debug();
-				$cursor = execute_query(
-					sub {
-						return get_products_collection(
-							get_products_collection_request_parameters($request_ref, {tags => 1}))
-							->aggregate($aggregate_parameters, {allowDiskUse => 1});
-					}
-				);
-			}
-			else {
-				$log->debug("Counting MongoDB documents for query", {query => $query_ref}) if $log->is_debug();
-				# test if query_ref is empty
-				if (single_param('no_count')) {
-					# Skip the count if it is not needed
-					# e.g. for some API queries
-					$log->debug("no_count is set, skipping count") if $log->is_debug();
-				}
-				elsif (keys %{$query_ref} > 0) {
-					#check if count results is in cache
-					my $key_count = generate_cache_key("search_products_count", $query_ref);
-					$log->debug("MongoDB query key - search_products_count", {key => $key_count}) if $log->is_debug();
-					my $results_count = get_cache_results($key_count, $request_ref);
-					if (not defined $results_count) {
-
-						$log->debug("count not in cache for query", {key => $key_count}) if $log->is_debug();
-
-						# Count queries are very expensive, if possible, execute them on the smaller products_tags collection
-						my $only_tags_filters = 1;
-
-						if ($server_options{producers_platform}) {
-							$only_tags_filters = 0;
-						}
-						else {
-
-							foreach my $field (keys %$query_ref) {
-								if ($field !~ /_tags$/) {
-									$log->debug(
-										"non tags field in query filters, cannot use smaller products_tags collection",
-										{field => $field, value => $query_ref->{field}}
-									) if $log->is_debug();
-									$only_tags_filters = 0;
-									last;
-								}
+					if (not defined $count) {
```

Comment:
It would be great to isolate this count  code in it's own function because it's a bit hard to read right now. There is more code for the count than for the mongodb search request and it's a bit lost !